ev3dev2/motor.py

```diff
@@ -1065,7 +1065,7 @@ def list_motors(name_pattern=Motor.SYSTEM_DEVICE_NAME_CONVENTION, **kwargs):
     """
     class_path = abspath(Device.DEVICE_ROOT_PATH + '/' + Motor.SYSTEM_CLASS_NAME)
 
-    return (Motor(name_pattern=name, name_exact=True)
```

Comment:
Why? I don't see any reason to believe these will always be `LargeMotor`s.

Reply:
This is so this can work on brickpi

Reply:
- It should be OK to instantiate a `Motor` on BrickPi -- it's only `MediumMotor` that doesn't work. A `Motor` can control a physical large motor. The only difference between the classes is the driver it filters for.
- The method here supports `MediumMotor`s too and has to work on other platforms. With this change `MediumMotor` is broken.

Reply:
I think we need to revisit the exception we are raising on brickpi for non-LargeMotor scenarios....will tackle that via a different issue.